src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -700,3 +706,34 @@ void uct_ib_mlx5_destroy_qp(uct_ib_mlx5_md_t *md, uct_ib_mlx5_qp_t *qp)
         break;
     }
 }
+
+ucs_status_t
+uct_ib_mlx5_iface_select_sl(const uct_ib_mlx5_iface_config_t *ib_mlx5_config,
+                            const uct_ib_iface_config_t *ib_config,
+                            const uct_iface_params_t *params,
+                            uct_ib_device_t *dev, uint8_t *sl_p)
+{
+    uint16_t ooo_sl_mask = 0;
+    uint8_t port_num;
+    ucs_status_t status;
+
+    status = uct_ib_device_find_port(dev, params->mode.device.dev_name,
```

Comment:
you can move it inside `uct_ib_iface_select_sl`, because it is common with verbs code.
Also looks like it is enough to pass `params->mode.device.dev_name` to `uct_ib_iface_select_sl`, then no need for finding a port

Reply:
we could not move it there, since we need to know the `port_num` for querying `ooo_sl_mask`